plutus-core/plutus-ir/test/TransformSpec.hs

```diff
@@ -88,6 +90,43 @@ letFloat =
   ,"regression1"
   ]
 
+recSplit :: TestNested
+recSplit =
+    testNested "recSplit"
+    $ map (goldenPir (RecSplit.recSplit . runQuote . PLC.rename) $ term @PLC.DefaultUni @PLC.DefaultFun)
+  [ "letInLet"
```

Comment:
I'd really rather have a set of tests designed for this pass than just copy the list of examples each time. This is useful for checking that nothing weird happens, but better to have tests that deliberately exercise the various cases.

Reply:
Are there any of these that actually exhibit the pass working?

Reply:
Some yes, see listMatch demoting and neweven